Dataflow program tile [#166986394]

kswenson opened the pull request:
- Show "CLUE" document workspace in Dataflow Workspace tab
- Add Dataflow program tile to Dataflow workspace

@mklewandowski This can reasonably be reviewed by commit, as the two are logically separate.

mklewandowski commented on src/index.tsx:
```diff
@@ -48,6 +49,7 @@ const problemOrdinal = urlParams.problem || DefaultProblemOrdinal;
 const {investigation, problem} = unit.getProblem(problemOrdinal) ||
                                  unit.getProblem(DefaultProblemOrdinal);
 const showDemoCreator = urlParams.demo;
+const createStores = urlParams.dataflow !== undefined ? createDataflowStores : createBaseStores;
```
So adding the `dataflow` URL parameter allows us to toggle between the Control Panel and a `DocumentWorkspaceComponent` and adding a URL parameter with `unit=dataflow` brings us to a `DocumentWorkspaceComponent` with modified left panels and tile toolbar.  In practice, how will these fit together (the `dataflow` and `unit` URL params)?

kswenson replied:
In the short term both are required to get the full effect. Ultimately, the authoring of the unit should support all of the desired variation, at which point the URL parameter can go away.

mklewandowski replied:
Got it.  I was using one at a time.  Not sure what impact that will have on my other comments.

mklewandowski commented on src/dataflow/components/dataflow-header.tsx:
```diff
@@ -38,7 +38,7 @@ export class DataflowHeaderComponent extends BaseComponent<IProps, {}> {
       <div className="dataflow-header">
         <div className="left">
-          DataFlow Application
+          Dataflow
```
These non-standard spellings are lurking everywhere!  Thanks for getting this!

mklewandowski commented on src/curriculum/dataflow/dataflow.json:
```diff
@@ -16,6 +16,12 @@
+    {
+      "name": "dataflow",
+      "title": "Dataflow",
+      "iconId": "icon-drawing-tool",
```
If we were feeling ambitious we could convert something from DF2 to an SVG icon to be used here.  This would work:
https://github.com/concord-consortium/flow-server/blob/master/static/flow/images/icon-program.png

Not very urgent though and probably better to do in a style pass.

mklewandowski commented on src/dataflow/models/tools/dataflow/dataflow-content.ts:
```diff
@@ -0,0 +1,21 @@
+  return DataflowContentModel.create();
+}
+
+export const kDataflowDefaultHeight = 480;
```
480 might be a little short.  However, the program editor div is also constrained to 500px at present.  In `dataflow-program.sass` we would also need to adjust the height of the `flow-tool` class.

kswenson replied:
Ultimately, we'll probably want some sort of use-all-available-space setting, but that's a story for another day.

mklewandowski commented:
Using both URL parameters together, I still see the React unique key prop error.  However, when I add a program tile, it no longer contains the Rete editor (although I can delete it).

kswenson commented:
- Fixed the unique key errors with a master PR which will be merged to the dataflow branch
- Fixed the issue with selection/deletion of the tile with an additional (one-line) commit to this PR
- I'll investigate the mobx errors you mentioned.

kswenson commented:
I'm not able to reproduce the mobx errors you report. The `hubStore` wouldn't exist without the `dataflow` URL parameter, so perhaps that occurred when running with just `unit=dataflow`?

kswenson commented:
>However, when I add a program tile, it no longer contains the Rete editor (although I can delete it).

The ability to delete the tile depends on the ability to select the tile. The tile is automatically selected upon creation, so I expect that explains the behavior of deletion. The selection issue should now be fixed. I have no explanation for the program tile not containing the Rete editor. I've never seen that behavior.

mklewandowski commented:
Mobx error only occurs if missing the `dataflow` param so it should be a moot point.
